contracts/proxy/Clones.sol

```diff
@@ -25,11 +25,12 @@ library Clones {
     function clone(address implementation) internal returns (address instance) {
         /// @solidity memory-safe-assembly
         assembly {
-            let ptr := mload(0x40)
-            mstore(ptr, 0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000)
-            mstore(add(ptr, 0x14), shl(0x60, implementation))
-            mstore(add(ptr, 0x28), 0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000)
-            instance := create(0, ptr, 0x37)
+            // Cleans the upper 96 bits of the `implementation` word, then packs the first 3 bytes
+            // of the `implementation` address with the bytecode before the address.
+            mstore(0x00, or(0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000, shr(0xe8, shl(0x60, implementation))))
+            // Packs the remaining 17 bytes of `implementation` with the bytecode after the address.
+            mstore(0x20, or(shl(0x78, implementation), 0x5af43d82803e903d91602b57fd5bf3))
+            instance := create(0, 0x09, 0x37)
```

Comment:
I'm not really sure why you are aligning to the end, and then starting the instance at 0x09. I would rather start at 0x00 like thid.
```suggestion
            mstore(0x00, or(0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000, shr(0x40, input)))
            mstore(0x20, or(0x00000000000000005af43d82803e903d91602b57fd5bf3000000000000000000, shl(0xc0, input)))
            instance := create(0, 0x00, 0x37)
```

Reply:
It’s to reduce the number of trailing zeroes in the constants. Smaller bytecode size.

Reply:
If the optimizer runs is low, the compiler may sometimes choose to break up constants with many trailing zeroes into multiple operations.

Reply:
ok, then we can do a different change optimization. (just updated the code above)

Reply:
The `shl` and `shr` is for cleaning the upper 96 bits. 

https://twitter.com/optimizoor/status/1560590413768994816?s=21&t=gSQglMPV2vg0zdLFwmzedA

Reply:
Hooo, right. I was assuming the address in clean